apps/dashboard/app/apps/ood_app.rb

```diff
@@ -181,19 +181,27 @@ def manifest_path
     path.join("manifest.yml")
   end
 
-  def icon_path_png
-    path.join("icon.png")
+  def svg_icon?
+    path.join("icon.svg").file?
   end
 
-  def icon_path_svg
-    path.join("icon.svg")
+  def png_icon?
+    path.join("icon.png").file?
   end
 
+  def icon_path
+    if svg_icon?
+      path.join("icon.svg")
+    elsif png_icon?
+      path.join("icon.png")
+    else
+      Pathname.new('')
+    end
+  end
+  
   def icon_uri
-    if icon_path_png.file? 
+    if svg_icon? || png_icon? 
```

Comment:
This test is in several places. Add an `OodApp#image_icon?` method so you can do `if image_icon?` and `if app.image_icon?` instead.